pkg/controllers/management/clusterstats/statsaggregator.go

```diff
@@ -225,3 +236,16 @@ func (s *StatsAggregator) machineChanged(key string, machine *v3.Node) (runtime.
 	}
 	return nil, nil
 }
+
+func isTaintedNoExecuteNoSchedule(m *v3.Node) bool {
+	for _, taint := range m.Spec.InternalNodeSpec.Taints {
+		if taint.Key != nodeRoleControlPlane && taint.Key != nodeRoleETCD &&
+			taint.Key != nodeRoleControlPlaneHyphen && taint.Key != nodeRoleMaster {
+			continue
+		}
```

Comment:
nit: feel like this is confusing conditional. You could do `notWorker := a || b || c` then `if notWorker && taint == NoSchedule; return true`

Reply:
@cbron Did something equivalent but without the inverted logic. Also, I try to avoid return the true case outside of conditionals as it's more idiomatic per: https://dave.cheney.net/2020/02/23/the-zen-of-go, but I think this reads fine now.